akka-stream/src/main/scala/akka/stream/Queue.scala

```diff
@@ -35,3 +43,29 @@ trait SinkQueue[T] {
    */
   def pull(): Future[Option[T]]
 }
+
+sealed abstract class StreamCallbackStatus[+T]
+
+/**
+ * Contains types that is used as return types for async callbacks to streams
+ */
+object StreamCallbackStatus {
```

Comment:
Hm, I am not sure if this needs to be so general. You can just have `QueueOfferResult` with various outcomes:
 - Enqueued
 - Dropped
 - ProtocolViolation("cannot do x while y")
 - QueueClosed